pallets/loans/src/interest.rs

```diff
@@ -0,0 +1,119 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use primitives::{Balance, CurrencyId};
+use sp_runtime::{traits::Zero, DispatchResult};
+
+use crate::*;
+
+impl<T: Config> Pallet<T> {
+    /// Accrue interest per block and update corresponding storage
+    pub(crate) fn accrue_interest() -> DispatchResult {
```

Comment:
maybe we can add https://substrate.dev/rustdocs/latest/frame_support/attr.require_transactional.html